scipy/sparse/linalg/_dsolve/linsolve.py

```diff
@@ -651,93 +649,94 @@ def spsolve_triangular(A, b, lower=True, overwrite_A=False, overwrite_b=False,
     Examples
     --------
     >>> import numpy as np
-    >>> from scipy.sparse import csr_matrix
+    >>> from scipy.sparse import csc_array
     >>> from scipy.sparse.linalg import spsolve_triangular
-    >>> A = csr_matrix([[3, 0, 0], [1, -1, 0], [2, 0, 1]], dtype=float)
+    >>> A = csc_array([[3, 0, 0], [1, -1, 0], [2, 0, 1]], dtype=float)
     >>> B = np.array([[2, 0], [-1, 0], [2, 0]], dtype=float)
     >>> x = spsolve_triangular(A, B)
     >>> np.allclose(A.dot(x), B)
     True
     """
 
     if is_pydata_spmatrix(A):
-        A = A.to_scipy_sparse().tocsr()
-
-    # Check the input for correct type and format.
-    if not (issparse(A) and A.format == "csr"):
-        warn('CSR matrix format is required. Converting to CSR matrix.',
+        A = A.to_scipy_sparse().tocsc()
+    
+    trans = "N"
+    if issparse(A) and A.format == "csr":
+        A = A.T
+        trans = "T"
+        lower = not lower
+
+    if not (issparse(A) and A.format == "csc"): 
+        warn('CSC or CSR matrix format is required. Converting to CSC matrix.',
              SparseEfficiencyWarning)
-        A = csr_matrix(A)
+        A = csc_matrix(A)
     elif not overwrite_A:
         A = A.copy()
 
-    if A.shape[0] != A.shape[1]:
+
+    M, N = A.shape
+    if M != N:
         raise ValueError(
             f'A must be a square matrix but its shape is {A.shape}.')
 
+    if unit_diagonal:
+        A = A.tolil()
+        A.setdiag(1)
+        A = A.tocsc()
+    else:
+        diag = A.diagonal()
+        if np.any(diag==0):
+            raise LinAlgError(
+                'A is singular: zero entry on diagonal.')
+        invdiag = 1/diag
+        if trans=="N":
+            A = A @ diags(invdiag)
+        else:
+            A = (A.T @ diags(invdiag)).T
+
     # sum duplicates for non-canonical format
     A.sum_duplicates()
-
+    
     b = np.asanyarray(b)
```

Comment:
```suggestion

    b = np.asanyarray(b)
```

Reply:
Fixed in 34af5470a5aafbcfd8c1d8af42c24e1e3ff0c5c8.

Reply:
This is still showing up under "requested changes", is there a way you could change that?

Reply:
@j-bowhay I think the requested changes have been made, but the CI box in the bottom still shows "Changes requested". Is there a way for you to change this?